lib/engine-addon.js

```diff
@@ -10,7 +10,9 @@ var concat = require('broccoli-concat');
 var DependencyFunnel = require('broccoli-dependency-funnel');
 var amdNameResolver = require('amd-name-resolver').moduleResolve;
 var defaultsDeep = require('lodash.defaultsdeep');
+var calculateCacheKeyForTree = require('calculate-cache-key-for-tree');
 
+var ADDON_TREE_CACHING = require('ember-cli/lib/experiments').ADDON_TREE_CACHING;
 var EOL = require('os').EOL;
```

Comment:
Needs to try/catch since `lib/experiments` isn't present on non-master ember-cli versions